test/support/tcp_proxy.rb

```diff
@@ -82,23 +82,39 @@ def forward(src, dst, pause_behavior: :ignore)
     zero_counter = 0
     loop do
       data = src.recv(1024)
-      if enabled?
+
+      if enabled? || pause_behavior == :ignore
         if data.empty?
           zero_counter += 1
           return if zero_counter >= 5
         else
           dst.send(data, 0)
         end
-      elsif pause_behavior == :return
+      elsif disabled? && pause_behavior == :return
         clean_data = data.encode("UTF-8", invalid: :replace, undef: :replace, replace: "")
-        warn "TCPProxy received a request while paused: `#{clean_data}`"
-        return
+
+        if schema_query?(clean_data)
```

Comment:
This is interesting! I remember we looked into that years ago and was probably related to which connection was used just before loading the schema